app/helpers/lti_helper.rb

```diff
@@ -0,0 +1,115 @@
+require 'mono_logger'
+module LtiHelper
+  # Synchronize LMS user with MarkUs users.
+  # if role is not nil, attempt to create users
+  # based on the values of can_create_users and
+  # can_create_roles.
+  def roster_sync(lti_deployment, course, can_create_users: false, can_create_roles: false)
+    errors = []
+    auth_data = lti_deployment.lti_client.get_oauth_token([LtiDeployment::LTI_SCOPES[:names_role]])
+    names_service = lti_deployment.lti_services.find_by!(service_type: 'namesrole')
+    membership_uri = URI(names_service.url)
+    membership_uri.query = URI.encode_www_form(role: 'http://purl.imsglobal.org/vocab/lis/v2/membership#Learner')
+    req = Net::HTTP::Get.new(membership_uri)
+    res = lti_deployment.send_lti_request!(req, membership_uri, auth_data, [LtiDeployment::LTI_SCOPES[:names_role]])
+    member_info = JSON.parse(res.body)
+    user_data = member_info['members'].filter_map do |user|
+      unless user['status'] == 'Inactive' || user['roles'].include?(LtiDeployment::LTI_ROLES['test_user'])
+        { user_name: user['lis_person_sourcedid'].nil? ? user['name'] : user['lis_person_sourcedid'],
+          first_name: user['given_name'],
+          last_name: user['family_name'],
+          display_name: user['name'],
+          email: user['email'],
+          lti_user_id: user['user_id'] }
+      end
+    end
+    if user_data.empty?
+      raise 'No user data returned from lms'
```

Comment:
If this is a message that will be visible to the end users, make it an I18n string